client/lib/mention.js

```diff
@@ -0,0 +1,71 @@
+var hueHash = require('./hue-hash')
+
+/**
+ * Determines if name contains the characters in the partial name, in order.
+ */
+module.exports.containsSubseq = function(name, part) {
+  var offset = 0
+  var remain = part
+  var nexdex
+  while (remain !== "") {
+    nexdex = name.indexOf(remain.substr(0, 1), offset)
+    if (nexdex < 0) {
+      return false
+    }
+    offset = nexdex + 1
+    remain = remain.substr(1)
```

Comment:
I don't see a way to do that and still use `indexOf`, unfortunately. I also wonder if JS does the same thing Java does where substring just creates a new pointer to a character array with an incremented *internal* pointer. :-)